### PR TITLE
#134 Add prompts for project names

### PR DESCRIPTION
**Pull request by bmonkman**

### CODE REVIEW

File: internal/context/init.go [hunk 1]
```diff
@@ -37,29 +39,45 @@ func Init(outDir string) *projectconfig.ZeroProjectConfig {
...
 	for k, v := range projectParameters {
 		projectConfig.Parameters[k] = v
 		// TODO: Add parameters to module structs inside project
 	}
 
+	for moduleName, _ := range moduleConfigs {
+		// @TODO : Uncomment when this struct is implemented
+		repoName := prompts[moduleName].GetParam(initParams)
+		repoURL := fmt.Sprintf("%s/%s", initParams["GithubRootOrg"], repoName)
+		//projectConfig.Modules[moduleName].Files.Directory = prompts[moduleName].GetParam(initParams)
```
Comment (bmonkman) [inline internal/context/init.go:76]: I'll uncomment this after Thiago's PR is merged.

File: internal/context/init.go [hunk 2]
```diff
@@ -70,8 +88,15 @@ func Init(outDir string) *projectconfig.ZeroProjectConfig {
 func loadAllModules(moduleSources []string) map[string]moduleconfig.ModuleConfig {
 	modules := make(map[string]moduleconfig.ModuleConfig)
 
+	wg := sync.WaitGroup{}
+	wg.Add(len(moduleSources))
+	for _, moduleSource := range moduleSources {
+		go module.FetchModule(moduleSource, &wg)
```
Comment (bmonkman) [inline internal/context/init.go:94]: Fetching will be done in parallel now

File: internal/context/init.go [hunk 3]
```diff
@@ -70,8 +88,15 @@ func Init(outDir string) *projectconfig.ZeroProjectConfig {
...
+	}
+	wg.Wait()
+
 	for _, moduleSource := range moduleSources {
-		mod, err := module.FetchModule(moduleSource)
+		mod, err := module.ParseModuleConfig(moduleSource)
```
Comment (bmonkman) [inline internal/context/init.go:99]: Split the parsing out from the fetching

File: internal/config/projectconfig/project_config.go [hunk 4]
```diff
@@ -9,10 +9,11 @@ import (
 )
 
 type ZeroProjectConfig struct {
-	Name           string
-	Infrastructure Infrastructure // TODO simplify and flatten / rename?
-	Parameters     map[string]string
-	Modules        []string
+	Name                   string
+	ShouldPushRepositories bool
+	Infrastructure         Infrastructure // TODO simplify and flatten / rename?
```
Comment (bmonkman) [inline internal/config/projectconfig/project_config.go:14]: Gonna remove this `Infrastructure` struct entirely in another PR

File: internal/context/init.go [hunk 5]
```diff
@@ -37,29 +39,45 @@ func Init(outDir string) *projectconfig.ZeroProjectConfig {
...
 		// TODO: Add parameters to module structs inside project
 	}
 
+	for moduleName, _ := range moduleConfigs {
+		// @TODO : Uncomment when this struct is implemented
+		repoName := prompts[moduleName].GetParam(initParams)
+		repoURL := fmt.Sprintf("%s/%s", initParams["GithubRootOrg"], repoName)
+		//projectConfig.Modules[moduleName].Files.Directory = prompts[moduleName].GetParam(initParams)
+		//projectConfig.Modules[moduleName].Files.Repository = repoURL
+		fmt.Println(repoURL)
```
Comment (davidcheung) [inline internal/context/init.go:78]: stray print command

File: internal/context/prompts.go [hunk 6]
```diff
@@ -15,17 +16,34 @@ import (
 type PromptHandler struct {
 	moduleconfig.Parameter
 	Condition func(map[string]string) bool
+	Validate  func(string) error
```
Comment (davidcheung) [inline internal/context/prompts.go:19]: 🎉  maybe later on we can implement some type of openapi schema validation, then user can supply that on the module side

File: internal/context/init.go (hunk 5 above)
Comment (bmonkman) [reply internal/context/init.go:78]: Yeah, reminder that that code needs to be filled in.

**Pull request merged**